ext/java/nokogiri/XmlSaxParserContext.java

```diff
@@ -94,7 +94,8 @@ protected void initialize(Ruby runtime) {
             parser = createParser();
         }
         catch (SAXException se) {
-            throw RaiseException.createNativeRaiseException(runtime, se);
+            // Unexpected failure in XML subsystem
+            throw runtime.newRuntimeError(se.getMessage());
```

Comment:
since this path was using the deprecated `NativeException` to wrap a Java throwable,
we could make it a bit closer to resembling the previous format (and set cause) :
```suggestion
            RaiseException ex = runtime.newRuntimeError(se.toString());
            ex.initCause(se);
            throw ex;
```
.. probably worth extracting to a helper for re-use.
also these changes might need inspecting caller and potentially using a custom wrapping Ruby exception if these were `rescue SAXException` anywhere.

Reply:
Yeah good call.